Mathlib/LinearAlgebra/RootSystem/Hom.lean

```diff
@@ -601,7 +633,31 @@ lemma reflection_coweightEquiv (P : RootPairing ι R M N) (i : ι) :
 
 @[simp]
 lemma reflection_indexEquiv (P : RootPairing ι R M N) (i : ι) :
-    (reflection P i).indexEquiv = P.reflection_perm i := rfl
+  (reflection P i).indexEquiv = P.reflection_perm i :=
+  rfl
+
+@[simp]
+lemma reflection_inv (P : RootPairing ι R M N) (i : ι) :
+    (reflection P i)⁻¹ = (reflection P i) := by
+  refine Equiv.ext ?_ ?_ ?_
+  · exact LinearMap.ext_iff.mpr (fun x => by simp [← weightEquiv_apply])
+  · exact LinearMap.ext_iff.mpr (fun x => by simp [← coweightEquiv_apply])
+  · exact _root_.Equiv.ext (fun j => by simp only [← indexHom_apply, map_inv]; simp)
```

Comment:
It occurs to me seeing this that we could make life easier here (and throughout this file) by proving stronger extensionality lemmas for `RootPairing.Hom` and `RootPairing.Equiv` (and dropping the `@[ext]` decorator from the `structure` declarations to free up the name).

I guess these extensionality lemmas would be heterogeneous versions of `RootPairing.weightHom_injective` and `RootPairing.coweightHom_injective` (which we would keep but with one-line proofs). IIUC these would read:
```lean
@[ext] protected lemma Hom.ext {x y : RootPairing.Hom P Q} (h : x.weightMap = y.weightMap) :
    x = y := by
  sorry

-- Cannot carry `ext` attribute but I guess worth having for symmetry.
protected lemma Hom.ext' {x y : RootPairing.Hom P Q} (h : x.coweightMap = y.coweightMap) :
    x = y := by
  sorry
```
Anyway, this is all a bit of a detour and need not be done now, I just thought I'd mention it.